File: server/internal/models/classification.go
package models

// Classification category values.
const (
	ClassActionRequired = "action_required"
	ClassNewsletter     = "newsletter"
	ClassFiltered       = "filtered"
	ClassTransactional  = "transactional"
)

// ClassifiedBy source values.
const (
	ClassifiedByRules    = "rules"
	ClassifiedByFeatures = "features"
	ClassifiedByLLM      = "llm"
	ClassifiedByUser     = "user"
)

// Classification holds the classification result for an email.
type Classification struct {
	Classification string  `json:"classification"`
	Confidence     float64 `json:"confidence"`
	ClassifiedBy   string  `json:"classified_by"`
	Reason         string  `json:"reason,omitempty"`
	IsOverridden   bool    `json:"is_overridden"`
}

// ReclassifyRequest is the request body for overriding an email's classification.
type ReclassifyRequest struct {
	NewClassification string `json:"new_classification"`
	Confirm           bool   `json:"confirm"`
}

// ValidClassifications returns all valid classification values.
func ValidClassifications() []string {
	return []string{ClassActionRequired, ClassNewsletter, ClassFiltered, ClassTransactional}
}

// ValidClassifiedBy returns all valid classified_by values.
func ValidClassifiedBy() []string {
	return []string{ClassifiedByRules, ClassifiedByFeatures, ClassifiedByLLM, ClassifiedByUser}
}

File: clients/shared/Sources/EmailClientKit/Models/Classification.swift
import Foundation

/// The classification category assigned to an email.
public enum ClassificationType: String, Codable, Sendable, Equatable, CaseIterable {
    case actionRequired = "action_required"
    case newsletter
    case filtered
    case transactional
}

/// Which layer of the classification pipeline assigned the classification.
public enum ClassifiedBy: String, Codable, Sendable, Equatable, CaseIterable {
    case rules
    case features
    case llm
    case user
}

/// Classification metadata for an email.
public struct Classification: Codable, Sendable, Equatable {
    public let classification: ClassificationType
    public let confidence: Double
    public let classifiedBy: ClassifiedBy
    public let reason: String?
    public let isOverridden: Bool?

    public init(
        classification: ClassificationType,
        confidence: Double,
        classifiedBy: ClassifiedBy,
        reason: String? = nil,
        isOverridden: Bool? = nil
    ) {
        self.classification = classification
        self.confidence = confidence
        self.classifiedBy = classifiedBy
        self.reason = reason
        self.isOverridden = isOverridden
    }
}
